components/insight/SRC/org/openmicroscopy/shoola/env/data/OmeroImageServiceImpl.java

```diff
@@ -1386,7 +1392,61 @@ public ScriptCallback createMovie(SecurityContext ctx, long imageID,
 			throw new IllegalArgumentException("No image specified.");
 		return gateway.saveROI(ctx, imageID, userID, roiList);
 	}
-	
+
+	/**
+     * Applies the transforms to the specified XML file.
+     *
+     * @param inputXML
+     *            The file to transforms.
+     * @param transforms
+     *            The collection of transforms.
+     * @param encoding The encoding to use.
+     * @return See above.
+     * @throws Exception
+     *             Thrown if an error occurred during the transformations.
+     */
+    private File applyTransforms(File inputXML, List<InputStream> transforms,
+            String encoding)
+            throws Exception {
+        TransformerFactory factory;
+        Transformer transformer;
+        InputStream stream;
+        Iterator<InputStream> i = transforms.iterator();
+        File output;
+        InputStream in = null;
+        OutputStream out = null;
+        try {
+            while (i.hasNext()) {
+                stream = i.next();
+                factory = TransformerFactory.newInstance();
+                Resolver resolver = new Resolver();
+                factory.setURIResolver(resolver);
+                output = File.createTempFile(
+                        RandomStringUtils.random(60, false, true),"."+XMLFilter.OME_XML);
+                output.deleteOnExit();
+                Source src = new StreamSource(stream);
+                Templates template = factory.newTemplates(src);
+                transformer = template.newTransformer();
+                transformer.setParameter(OutputKeys.ENCODING, encoding);
+                out = new FileOutputStream(output);
+                in = new FileInputStream(inputXML);
+                transformer.transform(new StreamSource(in),
+                        new StreamResult(out));
+                inputXML = output;
+                stream.close();
+                out.close();
+                in.close();
+                resolver.close();
+            }
+        } catch (Exception e) {
+            throw new Exception("Cannot apply transform", e);
+        }
+        File f = File.createTempFile(
```

Comment:
file is deleted in the finally.
I could certainly add a try catch finally in the loop to make sure  it is closed